ripd/ripd.c

```diff
@@ -1338,6 +1338,11 @@ static int rip_create_socket(struct rip *rip_param)
 	int sock;
 	struct sockaddr_in addr;
 
+	if (rip_param->vrf_id == VRF_UNKNOWN) {
+		zlog_err("VRF %s uninitialised", rip_param->name);
+		rip_param->sock = -1;
```

Comment:
this seems like we could have a case where a vrf is deleted out from under us and we just set this sock == -1 and then leak the fd.

Shouldn't the rip_create_socket function below be enough?